### PR TITLE
Add queries types

### PR DESCRIPTION
**Pull request by marcocastignoli**
See #1419

### CODE REVIEW

File: services/server/src/server/services/utils/database-util.ts [hunk 1]
```diff
@@ -140,18 +128,23 @@ export async function getVerifiedContractByChainAndAddress(
   );
 }
 
+export type GetSourcifyMatchByChainAddressResult = Tables.SourcifyMatch &
+  Pick<Tables.VerifiedContract, "creation_values" | "runtime_values"> &
```
Comment (manuelwedler) [inline services/server/src/server/services/utils/database-util.ts:132]: Nice how you used Pick to create a type for a joined query!

File: services/server/src/server/services/utils/database-util.ts [hunk 2]
```diff
@@ -607,7 +575,7 @@ export async function getSourcifyMatchAddressesByChainAndMatch(
   page: number,
   paginationSize: number,
   descending: boolean = false,
-) {
+): Promise<QueryResult<{ address: string }>> {
   let queryWhere = "";
   switch (match) {
     case "full_match": {
```
Comment (manuelwedler) [inline services/server/src/server/services/utils/database-util.ts:581]: (not related to this line, just couldn't pick the right one)

 The `updateContractDeployment` function would also need a return type.

**Review by manuelwedler — COMMENTED** [review]
Thanks for adding these! just a few small improvements that could be made

File: services/server/src/server/services/utils/database-util.ts (hunk 2 above)
Comment (marcocastignoli) [reply services/server/src/server/services/utils/database-util.ts:581]: it actually doesn't need a specific return type, I removed the `RETURN *`

Comment (manuelwedler) [reply services/server/src/server/services/utils/database-util.ts:582]: Okay great :+1:

**Review by manuelwedler — APPROVED** [review]

**Pull request merged**